app/assets/javascripts/discourse/components/text-field.js

```diff
@@ -11,6 +16,27 @@ export default TextField.extend({
     "dir"
   ],
 
+  didReceiveAttrs() {
+    this._super(...arguments);
+    this._prevValue = this.value;
+  },
+
+  didUpdateAttrs() {
+    this._super(...arguments);
+    if (this._prevValue !== this.value) {
+      if (this.onChangeImmediate) {
+        next(() => this.onChangeImmediate(this.value));
+      }
+      if (this.onChange) {
+        debounce(this, this._debouncedChange, DEBOUNCE_MS);
```

Comment:
This will throw an exception if you leave page during debounce timer

Reply:
I now cancel it, good catch.